### PR TITLE
[OSPP] The codes of Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Ianvs

### PR DESCRIPTION
**Pull request by Frank-lilinjie**
The PR of the proposal is in [PR](https://github.com/kubeedge/ianvs/pull/41)

This PR is merged in the branch named feature-lifelong-n.

This PR is about the codes of Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Ianvs.

This PR is a modification of the last [PR](https://github.com/kubeedge/ianvs/pull/38)

### CODE REVIEW

File: examples/curb-detection/lifelong_learning_bench/testalgorithms/rfnet/unseen_sample_recognition_by_scene.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        from torchvision import transforms
+        from PIL import Image
+        d_type = samples.data_type
+        x_data = samples.x
+        y_data = samples.y
+        data_transforms = transforms.Compose([
+                    transforms.Resize([32,32]),
+                    transforms.ToTensor()
+                    ])
+        #import pdb
```
Comment (luosiqi) [inline examples/curb-detection/lifelong_learning_bench/testalgorithms/rfnet/unseen_sample_recognition_by_scene.py:28]: Annotated codes should be removed.

File: examples/curb-detection/lifelong_learning_bench/testalgorithms/rfnet/unseen_sample_recognition_by_scene.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        #pdb.set_trace()
+        InferenceData=data_transforms(Image.open(x_data[0][0]))
+        device = 'cuda' if torch.cuda.is_available() else 'cpu'
+        pthfile = "./examples/curb-detection/lifelong_learning_bench/testalgorithms/rfnet/results/Epochofprose17.pth"
```
Comment (luosiqi) [inline examples/curb-detection/lifelong_learning_bench/testalgorithms/rfnet/unseen_sample_recognition_by_scene.py:32]: The value of "pthfile" should be set in __init__ function by a passed parameter.

File: examples/curb-detection/lifelong_learning_bench/testalgorithms/rfnet/unseen_sample_recognition_by_scene.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+from sedna.datasources import BaseDataSource
+from sedna.common.class_factory import ClassType, ClassFactory
+from sedna.algorithms.seen_task_learning.artifact import Task
+
+__all__ = ('UnseenSampleRecognitionByScene',)
+
+@ClassFactory.register(ClassType.UTD, alias="UnseenSampleRecognitionByScene")
+class UnseenSampleRecognitionByScene:
+    def __init__(self, **kwargs):
+        self.model_path = kwargs.get("model_path")
```
Comment (luosiqi) [inline examples/curb-detection/lifelong_learning_bench/testalgorithms/rfnet/unseen_sample_recognition_by_scene.py:12]: It seems that this parameter is not utilized.

File: examples/curb-detection/lifelong_learning_bench/testalgorithms/rfnet/unseen_sample_recognition_by_scene.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
+from typing import List, Any, Tuple
+
+from sedna.datasources import BaseDataSource
+from sedna.common.class_factory import ClassType, ClassFactory
+from sedna.algorithms.seen_task_learning.artifact import Task
+
+__all__ = ('UnseenSampleRecognitionByScene',)
+
+@ClassFactory.register(ClassType.UTD, alias="UnseenSampleRecognitionByScene")
+class UnseenSampleRecognitionByScene:
```
Comment (luosiqi) [inline examples/curb-detection/lifelong_learning_bench/testalgorithms/rfnet/unseen_sample_recognition_by_scene.py:10]: Add documentation comments to make this class more clarified.

File: examples/curb-detection/lifelong_learning_bench/testalgorithms/rfnet/rfnet_algorithm.yaml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    - type: "unseen_sample_recognition"
+      # name of python module; string type;
+      name: "UnseenSampleRecognitionByScene"
+      # the url address of python module; string type;
+      url: "./examples/curb-detection/lifelong_learning_bench/testalgorithms/rfnet/unseen_sample_recognition_by_scene.py"
+       # hyperparameters configuration for the python module; list type;
+      hyperparameters:
+        - model_path:
+            values:
+              - "./examples/curb-detection/lifelong_learning_bench/testalgorithms/rfnet/models/Epochofprose17.pth"
```
Comment (luosiqi) [inline examples/curb-detection/lifelong_learning_bench/testalgorithms/rfnet/rfnet_algorithm.yaml:69]: The model file "Epochofprose17.pth" has no origin which should be provided in readme.

**Comment by MooreZheng** [comment]
The code is overall fine but the conflict need to be resolved by fetching the latest version

**Comment by luosiqi** [comment]
lgtm

**Pull request merged**